whittle/prunning/pruners/base_pruner.py

```diff
@@ -0,0 +1,157 @@
+from argparse import Namespace
+from typing import Optional
+
+import torch
+import torch.nn as nn
+from torch.utils.data import DataLoader
+
+from whittle.models.gpt import GPT
+from whittle.modules.embedding import Embedding
+from whittle.modules.linear import Linear
+from whittle.prunning.utilis.catcher import Catcher
+
+
+class Pruner:
+    def __call__(
```

Comment:
I would just make this function abstract (i.e just call raise NotImplementError). Also I'd recommend to drop the args argument and just add a kwargs which can be parsed by the derived classes